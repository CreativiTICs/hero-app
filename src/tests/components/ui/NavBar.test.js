import React from "react";
import { mount } from "enzyme";
import { Navbar } from "../../../components/ui/NavBar";
import { MemoryRouter, Router } from "react-router-dom";

import { AuthContext } from "../../../components/auth/AuthContext";
import { types } from "../../../components/types/types";

describe("Pruebas en el <NavBar/>", () => {
  //Generar un history mock para probar el history
  const historyMock = {
    push: jest.fn(),
    replace: jest.fn(),
    location: {},
    listen: jest.fn,
    createHref: jest.fn,
  };

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: "Pedro",
    },
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter>
        {/* Le enviamos el historyMock al Router */}
        <Router history={historyMock}>
          <Navbar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  //Cuando se hace el mock se debe limpiar
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("Debe mostrarse correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".text-info").text().trim()).toBe("Pedro");
  });

  test("Debe llamar el logout y usar history", () => {
    wrapper.find("button").prop("onClick")();
    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.logout,
    });
    expect(historyMock.replace).toHaveBeenCalledWith("/login");
  });
});
