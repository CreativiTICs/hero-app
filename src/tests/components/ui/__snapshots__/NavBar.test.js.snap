// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <NavBar/> Debe mostrarse correctamente  1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href={[MockFunction]}
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        href={[MockFunction]}
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href={[MockFunction]}
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href={[MockFunction]}
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Pedro
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
